cosmos/launchpad/block_test.go

```diff
@@ -333,3 +333,54 @@ func TestLaunchpad_BlockTransaction(t *testing.T) {
 		},
 	}, tx)
 }
+
+func TestLaunchpad_BlockTransactionWithError(t *testing.T) {
+	mc := &cosmosmocks.CosmosTransactionsAPI{}
+	defer mc.AssertExpectations(t)
+
+	mc.
+		On("TxsHashGet", mock.Anything, "1").
+		Return(cosmosclient.TxQuery{
+			Txhash: "1",
+			Code:   7,
+			Tx: cosmosclient.StdTx{
+				Value: cosmosclient.StdTxValue{
+					Msg: []cosmosclient.Msg{
+						{
+							Type: "cosmos-sdk/MsgSend",
+							Value: cosmosclient.MsgValue{
+								FromAddress: "3",
+								ToAddress:   "6",
+								Amount: []cosmosclient.Coin{
+									{
+										Amount: "4",
+										Denom:  "5",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}, nil, nil).
+		Once()
+
+	properties := rosetta.NetworkProperties{
+		Blockchain: "TheBlockchain",
+		Network:    "TheNetwork",
+		SupportedOperations: []string{
+			"Transfer",
+			"Reward",
+		},
+	}
+
+	adapter := NewLaunchpad(TendermintAPI{}, CosmosAPI{Transactions: mc}, properties)
+
+	tx, _ := adapter.BlockTransaction(context.Background(), &types.BlockTransactionRequest{
```

Comment:
nit, would be nice to validate if errror is nil

Reply:
Hey, I skipped it because we already tested error (for the same behavior) in the tests above. I feel like it is best to not duplicate testing the same scenarios across different tests to make readability and maintenance easier. Wdyt?

Reply:
no problem, in this case is just a little require.Nil blah blah. So it is up to you, maybe adding it is not harmful in this case.

Reply:
IMHO I think the extra cost (1 extra line) is actually bearable